lib/optimizely.rb

```diff
@@ -46,8 +46,10 @@ class Project
     #                 By default all exceptions will be suppressed.
     # @param user_profile_service - Optional component which provides methods to store and retreive user profiles.
     # @param skip_json_validation - Optional boolean param to skip JSON schema validation of the provided datafile.
+    # @param config_manager - Optional Responds to get_config.
+    # @param notification_center - Optional Instance of NotificationCenter.
 
-    def initialize(datafile, event_dispatcher = nil, logger = nil, error_handler = nil, skip_json_validation = false, user_profile_service = nil)
+    def initialize(datafile = nil, event_dispatcher = nil, logger = nil, error_handler = nil, skip_json_validation = false, user_profile_service = nil, config_manager = nil, notification_center = nil)
```

Comment:
We should be able to also pass in the SDK key to use for instantiating a new `HTTPProjectConfigManager`